src/otx/core/model/entity/base.py

```diff
@@ -170,41 +173,67 @@ def map_class_names(src_classes: list[str], dst_classes: list[str]) -> list[int]
                 src2dst.append(-1)
         return src2dst
 
-    def export(self, output_dir: Path, export_format: OTXExportFormat) -> None:
+    def export(
+        self,
+        output_dir: Path,
+        export_format: OTXExportFormatType,
+        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
+    ) -> Path:
         """Export this model to the specified output directory.
 
         Args:
-            output_dir: Directory path to save exported binary files.
-            export_format: Format in which this `OTXModel` is exported.
+            output_dir (Path): directory for saving the exported model
+            export_format (OTXExportFormatType): format of the output model
+            precision (OTXExportPrecisionType): precision of the output model
+        Returns:
+            Path: path to the exported model.
         """
-        if export_format == OTXExportFormat.OPENVINO:
-            self._export_to_openvino(output_dir)
-        if export_format == OTXExportFormat.ONNX:
-            self._export_to_onnx()
-        if export_format == OTXExportFormat.EXPORTABLE_CODE:
-            self._export_to_exportable_code()
+        exporter = self._create_exporter()
```

Comment:
I think that you decided to create a `exporter` component to handle the export feature. I think it is good and will help us to enhance the bad cohesion about everything masses up in the base class. As things goes like this, I propose you to clean up everything in the base class and redesign it.

We will leave two abstract property in the base class such as
```python
class OTXModel:
    @property
    def exporter(self):
          msg = (
                "To export this OTXModel, you should implement an appropriate exporter for it. "
                "You can try to reuse our provided `Exporter` in `otx.core.exporter.*`."
          )
          raise NotImplementError(msg)

    @property
    def export_parameters(self) -> dict[str, Any]:
          msg = (
                "To export this OTXModel, you should define an appropriate parameters. "
                "This is used in the constructor of `self.exporter`. "
                "For example, `self.exporter = SomeExporter(**self.export_parameters)`. "
                "For the detail of constructor arguments, please see `otx.core.exporter.*`."
          )
          raise NotImplementError(msg)
```

Both are mandatory requirements to enable the model exportation that should be derived in the `OTXModel` children. With this design, we might maintain the common `export()` function in the base class such as
```python
    def export(
        self,
        output_dir: Path,
        export_format: OTXExportFormatType,
        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
    ) -> Path:
        return self.exporter.export(output_dir, export_format, precision)
```

Reply:
In fact, now export method is common and it's not supposed to be maintained. Also, abstract methods similar to the mentioned properties are already implemented. 
You're proposing to rewrite this with fancy syntaxis sugar, right?

Reply:
It's not just for the syntactic sugar. The real problem is https://github.com/openvinotoolkit/training_extensions/pull/2774#discussion_r1467185043. I saw your comment (https://github.com/openvinotoolkit/training_extensions/pull/2774#discussion_r1467741226), but I still don't know the point why they are split to the two sources although both are only used for the model exportation. 

> In brief,
metadata: structured, has a clear hierarchical structure, managed from the top of the hierarchy.
export parameters -> a murky dict of anything, created at low levels if needed. Now required only because of mm models.

Even if we make them as a single one, we still can have a hierarchical structure. Before you are telling about this in details, I could get what the rationale behind this although I'm very familiar with this codebase. Then, how about the others? I still think that it is natural that there is one source for the model exportation configuration. Lastly, I think this variable (`input_size`) in `export_params` is very model specific and not just for mm*, so that I don't think we should generalize it:
  https://github.com/openvinotoolkit/training_extensions/blob/74be80b044bf5e95778e46d94fbb1bbaa36c5120/src/otx/core/model/entity/classification.py#L218

Reply:
The example with input size you provided is valid only for classification mm model, which is almost the lowest level in the hierarchy, so it may not be generalizable. I implemented your proposal, (since we have no time for discussion), although I don't see how parameters became more clear to add. They are mixed with metadata now in an unstructured dict, which is not great, as I said.

Reply:
No, I meant that the input size is specific not just to the model architecture, furthermore to the model template (otx/recipe in our codebase). For example, you can imagine that there are two templates that train resnet50 on 320x320 or 224x224 sized images.